### PR TITLE
Chapter 4 and 5

### PR DESCRIPTION
**Pull request by mirulzaei**
ready

### CODE REVIEW

**Comment by mirulzaei** [comment]
ready

**Comment by manhnt-0960** [comment]
commented
It looks like you are missing content

**Comment by mirulzaei** [comment]
> commented
> It looks like you are missing content

is it because of the rename issue? I accidentally created a branch with different name and tried to rename it and then when i tried to push the first time it had errors but the 2nd time I had no problem with it.

**Comment by mirulzaei** [comment]
ready

**Comment by manhnt-0960** [comment]
commented
please merge 2 commits to 1

**Comment by mirulzaei** [comment]
ready

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * You're free to add application-wide styles to this file and they'll appear at the bottom of the
+ * compiled file so the styles you add here take precedence over styles defined in any other CSS/SCSS
+ * files in this directory. Styles in this file should be added after the last require_* statement.
+ * It is generally better to create a new file per style scope.
+ *
+ *= require_tree .
+ *= require_self
+ */
+
```
Comment (chungpv-1008) [inline app/assets/stylesheets/application.scss]: add endline for this file

File: app/assets/stylesheets/custom.scss [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  ul {
+    float: right;
+    list-style: none;
+
+    li {
+      float: left;
+      margin-left: 15px;
+    }
+  }
+}
```
Comment (chungpv-1008) [inline app/assets/stylesheets/custom.scss]: add endline for this file

File: app/controllers/static_pages_controller.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class StaticPagesController < ApplicationController
+  def home; end
+  def help; end
```
Comment (chungpv-1008) [inline app/controllers/static_pages_controller.rb]:
```suggestion
  def home; end

  def help; end
```

File: app/views/layouts/static_pages/home.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  <h2>
+    This is the home page for the
+    <%= link_to "Ruby on Rails Tutorial", Settings.links.rails_tutorial.home_page %>
+    sample application.
+  <h2>
+
+  <%= link_to "Sign up now!", signup_path, class: "btn btn-lg btn-primary" %>
+</div>
+
+<%= link_to image_tag("rails.svg", alt: "Rails logo", width: "200px"),"https://rubyonrails.org/" %>
```
Comment (chungpv-1008) [inline app/views/layouts/static_pages/home.html.erb]: add endline for this file

File: app/views/layouts/static_pages/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<%= provide :title, "Sign Up" %>
+
+<h1>Sign up</h1>
+<p>This will be a signup page for new users</p>
```
Comment (chungpv-1008) [inline app/views/layouts/static_pages/new.html.erb]: add endline for this file

File: config/routes.rb [hunk 6]
```diff
@@ -1,3 +1,9 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
```
Comment (chungpv-1008) [inline config/routes.rb]: remove this comment

File: config/settings.yml [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+  home_page: http://www.railstutorial.org/
+    help_page: http://www.railstutorial.org/help
+    book_page: http://www.railstutorial.org/book
+    contact_page: https://www.railstutorial.org/contact
+    rails_page: https://rubyonrails.org/
+    learn_enough: https://www.learnenough.com/
+    screencast_series: https://screencasts.railstutorial.org/
+    news: https://news.railstutorial.org/ 
+    michael_hartl: https://www.michaelhartl.com/
+
```
Comment (chungpv-1008) [inline config/settings.yml]: add endline for this file

**Comment by mirulzaei** [comment]
ready

File: app/controllers/_footer.html.erb [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<footer class="footer">
+  <small>
+	The <%= link_to "Ruby on Rails Tutorial", Settings.links.rails_tutorial.home_page %>
+	by <%= link_to "Michael Hartl", Settings.links.rails_tutorial.michael_hartl %>
+  </small>
```
Comment (saobv-2223) [inline app/controllers/_footer.html.erb]: indenting only 2 spaces

File: app/controllers/_footer.html.erb [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+  <nav>
+	<ul>
+	  <li><%= link_to "About", about_path %></li>
+	  <li><%= link_to "Contact", contact_path %></li>
+	  <li><%= link_to "News", Settings.links.rails_tutorial.news %>
+	</ul>
+  </nav>
```
Comment (saobv-2223) [inline app/controllers/_footer.html.erb]: indenting only 2 spaces

File: app/controllers/static_pages_controller.rb [hunk 10]
```diff
@@ -0,0 +1,10 @@
+class StaticPagesController < ApplicationController
+  def home; end
+
+  
+  def help; end
```
Comment (saobv-2223) [inline app/controllers/static_pages_controller.rb]:
```suggestion
  def home; end

  def help; end
```

File: app/views/layouts/application.html.erb [hunk 11]
```diff
@@ -1,16 +1,20 @@
...
     <%= stylesheet_link_tag "application", media: "all", "data-turbolinks-track": "reload" %>
     <%= javascript_pack_tag "application", "data-turbolinks-track": "reload" %>
+     <%= render "layouts/shim" %>
   </head>
 
   <body>
+   <%= render "layouts/header" %>
+  <div class="container">
     <%= yield %>
+     <%= render "layouts/footer" %>
```
Comment (saobv-2223) [inline app/views/layouts/application.html.erb]:
```suggestion
    <%= render "layouts/footer" %>
```

File: app/views/layouts/application.html.erb [hunk 12]
```diff
@@ -1,16 +1,20 @@
...
   <head>
-    <title>SampleApp</title>
-   <title><%= yield :title %> | Ruby on Rails Tutorial Sample App</title>
+   <title><%= yield :title %> | Ruby on Rails Tutorial </title>
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
     <%= stylesheet_link_tag "application", media: "all", "data-turbolinks-track": "reload" %>
     <%= javascript_pack_tag "application", "data-turbolinks-track": "reload" %>
+     <%= render "layouts/shim" %>
```
Comment (saobv-2223) [inline app/views/layouts/application.html.erb]:
```suggestion
    <%= render "layouts/shim" %>
```

File: app/views/layouts/static_pages/about.html.erb [hunk 13]
```diff
@@ -0,0 +1,11 @@
+<h1>About</h1>
+<p>
+
```
Comment (saobv-2223) [inline app/views/layouts/static_pages/about.html.erb]:
```suggestion
<p>
```

File: app/views/layouts/static_pages/help.html.erb [hunk 14]
```diff
@@ -0,0 +1,5 @@
+ <%= link_to "Rails Tutorial help page", Settings.links.rails_tutorial.help_page %>
+
```
Comment (saobv-2223) [inline app/views/layouts/static_pages/help.html.erb:2]:
```suggestion
```

**Comment by mirulzaei** [comment]
ready

File: app/views/layouts/static_pages/help.html.erb [hunk 15]
```diff
@@ -0,0 +1,4 @@
+ <%= link_to "Rails Tutorial help page", Settings.links.rails_tutorial.help_page %>
+  To get help on this sample app, see the
+  <%= link_to "Ruby on Rails Tutorial book", Settings.links.rails_tutorial.book_page %>
+</p>
```
Comment (saobv-2223) [inline app/views/layouts/static_pages/help.html.erb:8]:
```suggestion
<p>
 <%= link_to "Rails Tutorial help page", Settings.links.rails_tutorial.help_page %>
  To get help on this sample app, see the
  <%= link_to "Ruby on Rails Tutorial book", Settings.links.rails_tutorial.book_page %>
</p>
```

Comment (mirulzaei) [reply app/views/layouts/static_pages/help.html.erb:8]: done

**Comment by mirulzaei** [comment]
ready

**Comment by manhnt-0960** [comment]
merged

**Pull request merged**